components/org.wso2.carbon.identity.account.suspension.notification.task/src/main/java/org/wso2/carbon/identity/account/suspension/notification/task/AccountValidatorThread.java

```diff
@@ -17,41 +17,106 @@
 import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.governance.IdentityGovernanceException;
 import org.wso2.carbon.identity.mgt.services.UserIdentityManagementAdminService;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.user.core.tenant.Tenant;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
 
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 public class AccountValidatorThread implements Runnable {
 
     private static final Log log = LogFactory.getLog(AccountValidatorThread.class);
 
-    private AccountValidatorThreadProperties accountValidatorThreadProperties = null;
+    public AccountValidatorThread() {
 
-    public AccountValidatorThread(AccountValidatorThreadProperties accountValidatorThreadProperties) {
-        this.accountValidatorThreadProperties = accountValidatorThreadProperties;
     }
 
-    public AccountValidatorThread() {
-    }
+    @Override
+    public void run() {
+
+        RealmService realmService = NotificationTaskDataHolder.getInstance().getRealmService();
+
+        Tenant tenants[] = new Tenant[0];
+
+        try {
+            tenants = (Tenant[]) realmService.getTenantManager().getAllTenants();
+        } catch (UserStoreException e) {
+            log.error("Error occurred while retrieving tenants", e);
+        }
+
+        handleRun(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
 
-    @Override public void run() {
-        if (log.isDebugEnabled()) {
-            log.debug("Executing the scheduled task...");
+        for(Tenant tenant:tenants) {
+            handleRun(tenant.getDomain());
         }
 
-        notifyUsers();
+    }
+
+    private void handleRun(String tenantDomain) {
 
+        Property[] identityProperties;
         try {
-            disableAccounts();
+            identityProperties = NotificationTaskDataHolder.getInstance().getIdentityGovernanceService()
+                    .getConfiguration(getPropertyNames(), tenantDomain);
+            boolean isEnabled = false;
+            long suspensionDelay = 0;
+            long[] notificationDelays = null;
+            for (Property identityProperty : identityProperties) {
+
+                if (identityProperty == null) {
+                    continue;
+                }
+
+                if (NotificationConstants.SUSPENSION_NOTIFICATION_ENABLED.equals(identityProperty.getName())) {
+                    isEnabled = Boolean.parseBoolean(identityProperty.getValue());
+                }
+
+                if (NotificationConstants.SUSPENSION_NOTIFICATION_ACCOUNT_DISABLE_DELAY.
+                        equals(identityProperty.getName())) {
+                    suspensionDelay = Long.parseLong(identityProperty.getValue());
```

Comment:
Need to handle NumberFormatException. Its ok to leave processing for this perticular tenant but avoid breaking functionality for other tenants.